### ISSUE TITLE
Gather feedback on NEW resume review portal

### ISSUE THREAD

**Issue by peironggg** [yangshun/tech-interview-handbook#301]
We are a group of 4 students from Singapore that is thinking of contributing a resume review portal to the handbook.

The portal will be an extension of Tech Interview Handbook (TIH). It will allow software engineers to get feedback on their resumes from the community. Users will upload their resume for review and voluntary reviewers can leave their comments on different sections of the resume.

Below are our proposed features, feel free to share any feedback you feel will improve the benefit of using this portal for both reviewers and uploaders!

For the resume review portal, we have 3 key functionalities:
- **Submission**: For users to submit their resumes to be reviewed by others.
    - Submission Form
        - Title
        - Role
        - Experience Level
        - Location
        - Description/Additional Info

![Screenshot 2022-09-29 at 7 29 37 PM](https://user-images.githubusercontent.com/35712975/193020142-b308858b-3584-4eeb-b3f1-de2d6a7e1878.png)

![Screenshot 2022-09-29 at 7 30 02 PM](https://user-images.githubusercontent.com/35712975/193020212-0abfee39-4a92-438f-b95b-68b6ea3a7b4e.png)

- **Review**: For users to review submitted resumes and add comments.
    - Page to view a resume and see existing reviews / comments
    - Pre-defined sections for users to review
    - Proposed sections: Contact Info/Education/Skills/Experience/Projects/General comments
  
![Screenshot 2022-09-29 at 7 30 38 PM](https://user-images.githubusercontent.com/35712975/193020325-641e6a96-cd2a-432e-9068-6723f84c8a36.png)

- **Browse**: A user can browse submitted and reviewed resumes.
    - Sorting of resumes (by recency, number of comments etc.)
    - Filters based on role, experience and location
    - Search based on keywords in Title/Description

![Screenshot 2022-09-29 at 7 31 01 PM](https://user-images.githubusercontent.com/35712975/193020388-8b9a891f-cd58-45c3-b492-84bd8a1a50b3.png)
![Screenshot 2022-09-29 at 7 31 21 PM](https://user-images.githubusercontent.com/35712975/193020452-2e89a8a0-c7ce-4edd-bb5c-7be82c03742a.png)

**Comment by xuan-nguyen-swe** [comment]
I guess the design only shows the original PDF on the left as it is, without parsing, right?
Is the section on the right (Education, Experience) hard code or dynamic?
Where is the database and file storage for the portal?

**Comment by peironggg** [comment]
> I guess the design only shows the original PDF on the left as it is, without parsing, right? 
Yep, we will leave the resume as is so the reviewer can also see the style, font, etc of the original resume and provide feedback in those areas

> Is the section on the right (Education, Experience) hard code or dynamic?
We will leave those sections hardcoded for now but going into the future, we aim to make it dynamic. It will require some tinkering with PDF.js to get the sections right all the time since parsing PDFs can be a tricky affair.

> Where is the database and file storage for the portal?
We are using PostgreSQL to store the resume metadata (date uploaded, etc) and comments. We are still thinking over the options for storing the resume, whether we store it in the database as well or find a dedicated storage provider.

Hope we answered your questions :)

**Comment by tonghuikang** [comment]
The interface is minimal and neat. I am happy to see that the jobseeker does not need to manually fill up a bunch of forms before getting comments (unlike applying for jobs). If you want some technical depth in the implementation you can also consider designing some heuristics to decide which resumes to show in the front page and in what order.

Similar current solutions include posting on Reddit ([generic resumes](https://www.reddit.com/r/resumes/), [generic engineering](https://www.reddit.com/r/EngineeringResumes/), [tech resumes](https://www.reddit.com/r/cscareerquestions/search?q=author%3ACSCQMods%20Resume%20Advice%20Thread))

Some questions and proposals
- Will your portal support only tech resumes? Since this is the Tech Interview Handbook, probably we can just focus on tech resumes.
- Will your portal support only one-page resumes? I think we can support only one-page resumes. I heard comments that recruiters and hiring managers strongly prefer one-page resumes to the extent that they will ignore multiple-page resumes for the inability to summarize.
- Will your portal support the ability to copy text from the resume? It will make it easier to cite and comment.
- Probably you want to do some user study on whether commenters prefer to break their comments over different sections. I would prefer to keep all my comments in one place, it seems to work on Reddit.

**Comment by peironggg** [comment]
> Will your portal support only tech resumes? Since this is the Tech Interview Handbook, probably we can just focus on tech resumes.

Yep!

> Will your portal support only one-page resumes? I think we can support only one-page resumes. I heard comments that recruiters and hiring managers strongly prefer one-page resumes to the extent that they will ignore multiple-page resumes for the inability to summarize.

We plan to give them the flexibility to have more than 1 page since the standards may differ based on different locations they are applying to. It helps to accommodate our users who come from different parts of the world.

> Will your portal support the ability to copy text from the resume? It will make it easier to cite and comment.

Yep! We can make it work with PDF.js

> Probably you want to do some user study on whether commenters prefer to break their comments over different sections. I would prefer to keep all my comments in one place, it seems to work on Reddit.

This a very good suggestion! Thank you :)
